pyincore/utils/popdisloutputprocess.py

```diff
@@ -46,31 +46,36 @@ def __init__(self, pop_disl_result, pop_disl_result_path=None,
                 pd_result_flag = pd_result[(pd_result["guid"].notnull()) &
                                            (pd_result["numprec"].notnull()) &
                                            (pd_result["plcname10"] == filter_name)]
-                # only keep guid, place and dislocated
-                pd_result_shp = pd_result[(pd_result["dislocated"]) &
-                                          (pd_result["guid"].notnull()) &
+                # only keep guid, place
+                pd_result_shp = pd_result[(pd_result["guid"].notnull()) &
                                           (pd_result["numprec"].notnull()) &
                                           (pd_result["plcname10"] == filter_name)]
+                # Update "numprec" to 0 where "dislocated" is False
+                pd_result_shp.loc[pd_result_shp["dislocated"] == False, "numprec"] = 0
```

Comment:
I feel like you might be able to pull this ` pd_result_shp.loc[pd_result_shp["dislocated"] == False, "numprec"] = 0` outside all the if statement; because this should apply to everything regardless of the criteria?

Reply:
== False --> 
Could you check the `pd_result_shp["dislocated"]` is it indeed a boolean?

Reply:
Thanks for catching this! pd_result_shp["dislocated"] is a boolean and shouldn't compare with False.

Reply:
moved `pd_result_shp.loc[pd_result_shp["dislocated"] == False, "numprec"] = 0` outside of all if statement, good suggestion!
set numprec as integer as well